### PR TITLE
Feature/replace fftw complex

### PR DESCRIPTION
**Pull request by MCFlowMace**

### CODE REVIEW

**Review by nsoblath — CHANGES_REQUESTED** [review]
This is an impressive set of changes.  Clearly it cleans things up nicely.

There are several new dependencies on Eigen, including KTFractionalFFT and KTPhysicalArrayComplex (well, the latter I guess came in the previous PR).  We should either adopt Eigen as a standard dependency (which is the way it's headed anyhow) or modify the appropriate CMakeLists.txt files to only build/use the affected files if eigen is found.  I'm inclined to go with the former.

File: Source/Data/Transform/KTFrequencySpectrumFFTW.cc [hunk 1]
```diff
@@ -61,45 +61,50 @@ namespace Katydid
         //KTINFO(fslog, "neg freq offset: " << fLeftOfCenterOffset);
     }
 
+    // I have doubts about this function
     KTFrequencySpectrumFFTW::KTFrequencySpectrumFFTW(std::initializer_list<double> value, size_t nBins, double rangeMin, double rangeMax, bool arrayOrderIsFlipped) :
```
Comment (MCFlowMace) [inline Source/Data/Transform/KTFrequencySpectrumFFTW.cc:65]: @nsoblath I do not think that this function does what it is supposed to do. Either that, or I do not understand its purpose. From the signature it looks like it should enable initialization like 
`KTFrequencySpectrumFFTW s({0.0, 1.0, 2.0, 3.0}, 2, 0.0, 1.0);`
and that should result in a 2 bin spectrum with s[0] = 0.0 + 1.0i, s[1] = 2.0 + 3.0i. If that is the case, it is not what the function does.

File: Source/Data/Transform/KTFrequencySpectrumFFTW.cc [hunk 2]
```diff
@@ -61,45 +61,50 @@ namespace Katydid
...
-        fIsArrayOrderFlipped = rhs.fIsArrayOrderFlipped;
-        fIsSizeEven = rhs.fIsSizeEven;
-        fLeftOfCenterOffset = rhs.fLeftOfCenterOffset;
-        fCenterBin = rhs.fCenterBin;
-        fConstBinAccess = rhs.fConstBinAccess;
-        fBinAccess = rhs.fBinAccess;
-        fNTimeBins = rhs.fNTimeBins;
-        return *this;
-    }
+    //copy constructor, destructor and copy assignment operator
```
[the remark this answers is not in the document]
Comment (MCFlowMace) [reply Source/Data/Transform/KTFrequencySpectrumFFTW.cc]: Done

**Comment by MCFlowMace** [comment]
At the moment the optional `Katydid_USE_EIGEN` flag in the topmost CMakeLists.txt is just turned on by default. But can we make it optional still after the standard spectrum and timeseries objects now rely on it? What do you suggest to change other than switching the flag on by default?
Or maybe the question is actually, how is a standard dependency handled differently?

**Review by psurukuc — APPROVED** [review]

**Comment by psurukuc** [comment]
Other than the ongoing disucssions about Eigen3 in CMakeLists, it all looks good to me.

**Review by nsoblath — APPROVED** [review]

**Comment by nsoblath** [comment]
@MCFlowMace is there a minimum Eigen version?

**Comment by MCFlowMace** [comment]
> @MCFlowMace is there a minimum Eigen version?

I developed with version 3.3.9, might also work with earlier versions of course but that's untested.

**Pull request merged**